src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -888,22 +888,22 @@ namespace cryptonote
 
 
     {
-      m_core.pause_mine();
-      epee::misc_utils::auto_scope_leave_caller scope_exit_handler = epee::misc_utils::create_scope_leave_handler(
-        boost::bind(&t_core::resume_mine, &m_core));
-
       MLOG_YELLOW(el::Level::Debug, "Got NEW BLOCKS inside of " << __FUNCTION__ << ": size: " << arg.blocks.size());
 
       if (m_core.get_test_drop_download() && m_core.get_test_drop_download_height()) { // DISCARD BLOCKS for testing
 
-        uint64_t previous_height = m_core.get_current_blockchain_height();
-
         // we lock all the rest to avoid having multiple connections redo a lot
         // of the same work, and one of them doing it for nothing: subsequent
         // connections will wait until the current one's added its blocks, then
         // will add any extra it has, if any
         CRITICAL_REGION_LOCAL(m_sync_lock);
 
+        m_core.pause_mine();
```

Comment:
Sorry @moneromooo-monero this just leads to more questions. But I do think its worth getting this betterish.

I wasn't sure what `miner::pause()`/`miner::resume()` did previously (thus my questions), and you accurately responded about the reference count. So the current behavior (before this patch) is that the miner _may_ remain paused across multiple responses, whereas this patch always `pause()`/`resume()` for every processed response. Your comment in the PR description indicates otherwise - did you mean the mining threads won't immediately start again?

Either way, it seems like pausing should be done after the `empty()` check. But then a `goto` wouldn't be possible because it would skip the construction of `scope_exit_handler`, and placing that variable into its own scope means resuming before `request_missing_objects(...)` is called. I think this is valid, but crap.